### DISCUSSION TITLE
Where to find help

### DISCUSSION THREAD

**Discussion by DanBee0815** [rectorphp/rector#7260] [Q&A] [answered]
Hello Folks,

im trying to upgrade a very old php5 site to run under php8. I got rector installed with composer under Win10.
I followed the tiny tutorials given and setup the minimum configuration which is provided in here.

It kinda works: My testfile which only includes an echo "Hello World!", is getting processed, but then quits with:
[Error] Could not process some files, due to:
"Child process error: " . 

Yes, there is no error attached (after the dot), it just quits...

My rector.php looks like this:

`use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;

return static function (RectorConfig $rectorConfig): void {
    // A. run whole set
    $rectorConfig->sets([
        LevelSetList::UP_TO_PHP_81
    ]);

};`

So, my Question is: Where do I start to find a solution? Is there a Forum, Stackoverflow (the Rector section is tiny and dead it seems), where I could read a bit more or get more examples of the general setup?

Cheers,
Dan

**Comment by samsonasik** [comment, marked as answer]
That's seems parallel process, could you try with disable parallel:

```php
$rectorConfig->disableParallel();
```

and see if it is working? 

I don't have windows10 at my local dev, so can't test the exact environment. 

We need the exact step by step to reproduce the issue, (the file, the config, the command that run), better to use Github repository with github action running.

You may can create reproducible repository with Github action with windows for it https://github.blog/changelog/2022-01-11-github-actions-jobs-running-on-windows-latest-are-now-running-on-windows-server-2022/

**Reply by DanBee0815** [reply]
Hey Abdul,

thank you VERY much for your kind and fast answer - much appriciated!

Using $rectorConfig->disableParallel(); within the rector.php did the trick. Also it's no Problem to just wait a little longer until the processing is finished in single threaded mode.

What I would really love to know is where did you get that information from? To me it seems I did not yet understand where to get more info about rector than on the /rectorphp/rector/tree/main/docs page. 

But where could I find Information about specific levelSetLists for example? Im asking because I ran the LevelSetList::UP_TO_PHP_81 on the said site, but it still has the same amount of notices as it had before and I'd like to look for what I missed out on configuring / choosing what to process.


Thank you very much,
Dan

**Reply by samsonasik** [reply]
You can get list of functions from this class:

https://github.com/rectorphp/rector/blob/6f73a90692f25f2a245a44579580899fa0b8d3d4/packages/Config/RectorConfig.php#L19

That's seems missing documentation, feel free to add new documentation PR to:

https://github.com/rectorphp/rector-src/tree/main/build/target-repository/docs

what notice? usually, updating very old project need step by step, not a big collection of set list, so you can see the diff and verify you can start with up to php 7.x, eg:

```php
<?php

use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;

return static function (RectorConfig $rectorConfig): void {
   $rectorConfig->sets([
        LevelSetList::UP_TO_PHP_70
    ]);
};
```

if it is working, continue to `LevelSetList::UP_TO_PHP_71`, and so on. You can get list of SetList from here:

https://github.com/rectorphp/rector/tree/6f73a90692f25f2a245a44579580899fa0b8d3d4/packages/Set/ValueObject
